### PR TITLE
Update dependencies in 3.0.x branch to match those in rasa 3.0.x

### LINKED ISSUE #592: Update `rasa-sdk` dependencies to align with those of `rasa`
**Description of Problem**:

Installing editable version of `rasa-sdk` after having installed editable version of `rasa` in the same venv causes the bot to crash when running `rasa shell`, full stack trace below:

```
Bot loaded. Type a message and press enter (use '/stop' to exit): 
2021-11-11 13:26:28 ERROR    asyncio  - Task exception was never retrieved
future: <Task finished name='Task-6' coro=<SignalRouter._dispatch() done, defined at /Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py:102> exception=RuntimeError('this event loop is already running.') created at /Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py:165>
source_traceback: Object created at (most recent call last):
  File "/Users/ancalita/rasa-projects/rasa-oss-09/bin/rasa", line 5, in <module>
    main()
  File "/Users/ancalita/rasa-projects/rasa/rasa/__main__.py", line 121, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/Users/ancalita/rasa-projects/rasa/rasa/cli/shell.py", line 125, in shell
    rasa.cli.run.run(args)
  File "/Users/ancalita/rasa-projects/rasa/rasa/cli/run.py", line 126, in run
    rasa.run(**vars(args))
  File "/Users/ancalita/rasa-projects/rasa/rasa/api.py", line 56, in run
    rasa.core.run.serve_application(
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/run.py", line 221, in serve_application
    app.run(
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1051, in run
    serve_single(server_settings)
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/server/runners.py", line 199, in serve_single
    serve(**server_settings)
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/server/runners.py", line 146, in serve
    loop.run_until_complete(app._server_event("init", "after"))
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1496, in _server_event
    await self.dispatch(
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 163, in dispatch
    return await dispatch
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 133, in _dispatch
    retval = await maybe_coroutine
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1382, in _listener
    await maybe_coro
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1368, in dispatch_delayed_tasks
    await app.dispatch(name, context={"app": app, "loop": loop})
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 165, in dispatch
    task = asyncio.get_running_loop().create_task(dispatch)
Traceback (most recent call last):
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 133, in _dispatch
    retval = await maybe_coroutine
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1374, in run_delayed_task
    await prepped
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/run.py", line 134, in run_cmdline_io
    await console.record_messages(
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/channels/console.py", line 175, in record_messages
    text = _get_user_input(previous_response)
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/channels/console.py", line 108, in _get_user_input
    response = questionary.text(
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/questionary/question.py", line 50, in ask
    return self.unsafe_ask(patch_stdout)
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/questionary/question.py", line 64, in unsafe_ask
    return self.application.run()
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 890, in run
    return loop.run_until_complete(
  File "uvloop/loop.pyx", line 1450, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1443, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1351, in uvloop.loop.Loop.run_forever
  File "uvloop/loop.pyx", line 480, in uvloop.loop.Loop._run
```

This [forum thread](https://forum.rasa.com/t/error-this-event-loop-is-already-running/24017/9) seems to pin the cause of this issue to the version of `questionary` which I noticed `rasa-sdk` downgrades from `1.10` to `1.8` when running `make install`.

**Definition of Done**:
- [ ] `questionary` version in `rasa-sdk` is pinned to the higher version as in rasa-oss
- [ ] double-check all dependencies are pinned to the higher version pinned in rasa-oss
- [ ] you are able to run `rasa shell` without issues when both projects are installed as editable in the same venv

### PR DESCRIPTION
**Pull request by carlad**
Partially closes #592 

I used rasa as the primary dependency source, so most changes were made in this PR.
There was one change made in `rasa`, as there was a python 3.9 compatibility issue with `black`. That PR can be found here: [https://github.com/RasaHQ/rasa/pull/10667](https://github.com/RasaHQ/rasa/pull/10667)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -63,24 +63,24 @@ name = "Miscellaneous internal changes"
 showcontent = false
 
 [tool.poetry.dependencies]
-python = ">=3.7,<3.10"
+python = ">=3.7,<3.9"
 coloredlogs = ">=10,<16"
 sanic = "^21.6.0"
-requests = ">=2.23.0,<2.26.0"
+requests = ">=2.23"
```
Comment (ancalita) [inline pyproject.toml:69]: In `rasa`, this is pinned to `^2.23`, could you please update here too?

**Review by usc-m — APPROVED** [review]
LGTM

**Review by joejuzl — APPROVED** [review]
💯

**Review by ancalita — APPROVED** [review]
👍🏼

**Pull request merged**